### PR TITLE
Add documentation to support RNDIS features

### PR DESCRIPTION
**Pull request by ajpenner**
**- What I did**
Added documentation to explain the build variables for RNDIS support

**- How I did it**
I modified the extra-build-config.md document to include a brief description of the variables. I have also added two new variables to allow the user to supply two MAC addresses for the host and device and add them to the cmdline.txt file

### CODE REVIEW

File: recipes-bsp/bootfiles/rpi-cmdline.bb [hunk 1]
```diff
@@ -34,7 +34,13 @@ CMDLINE_DEBUG ?= ""
 # CMDLINE_RNDIS = "modules-load=dwc2,g_ether g_ether.host_addr=<some MAC 
 # address> g_ether.dev_addr=<some MAC address>"
 # if the MAC addresses are omitted, random values will be used
-CMDLINE_RNDIS ?= ""
+CMDLINE_RNDIS ?= '${@oe.utils.conditional("ENABLE_RNDIS", "1", "modules-load=dwc2,g_ether", "", d)}'
```
Comment (agherzan) [inline recipes-bsp/bootfiles/rpi-cmdline.bb:37]: `modules-load` is systemd-specific, right?

File: recipes-bsp/bootfiles/rpi-cmdline.bb [hunk 2]
```diff
@@ -34,7 +34,13 @@ CMDLINE_DEBUG ?= ""
...
+
+TMP_HOST_MAC_ADDR = '${@oe.utils.conditional('RNDIS_HOST_MAC_ADDR', None, "", " g_ether.host_addr=" + str(d.getVar('RNDIS_HOST_MAC_ADDR')), d)}'
+CMDLINE_RNDIS_HOST_MAC_ADDR ?= '${@oe.utils.conditional("ENABLE_RNDIS", "1", TMP_HOST_MAC_ADDR, "", d)}'
```
Comment (agherzan) [inline recipes-bsp/bootfiles/rpi-cmdline.bb:40]: Would it be cleaner to use an anon function to handle this? It gets a bit tangled. Same for the set below.

File: recipes-bsp/bootfiles/rpi-cmdline.bb (hunk 1 above)
Comment (agherzan) [inline recipes-bsp/bootfiles/rpi-cmdline.bb:37]: Isn't this dependent on the kernel configuration? module/built-in that is.

**Review by agherzan — CHANGES_REQUESTED** [review]
Besides the comments I've raised, I would like to have the branch rebase and the history cleaned up (it would be nice to have one commit for docs and one for cmdline - but I wouldn't be too strict about it).

File: recipes-bsp/bootfiles/rpi-cmdline.bb (hunk 1 above)
Comment (agherzan) [reply recipes-bsp/bootfiles/rpi-cmdline.bb:37]: In that case this should also have a check on systemd.

Comment (ajpenner) [reply recipes-bsp/bootfiles/rpi-cmdline.bb:37]: Yes it does, this requires that the OTG setting is also enabled. I will elaborate the dependencies to clarify this.

File: recipes-bsp/bootfiles/rpi-cmdline.bb (hunk 2 above)
Comment (ajpenner) [reply recipes-bsp/bootfiles/rpi-cmdline.bb:40]: Great point

Comment (ajpenner) [reply recipes-bsp/bootfiles/rpi-cmdline.bb]: Done

File: recipes-bsp/bootfiles/rpi-cmdline.bb (hunk 1 above)
Comment (ajpenner) [reply recipes-bsp/bootfiles/rpi-cmdline.bb]: Done

**Comment by ajpenner** [comment]
I will realign my PR to make two as you suggest. I agree it is cleaner (but ran out of time today)

**Comment by ajpenner** [comment]
Okay, I believe everything is in order for another review or accepting the PR.

Thanks!

**Pull request closed without merging**